projects/plugins/jetpack/modules/infinite-scroll/infinity.php

```diff
@@ -779,7 +780,7 @@ public function query_time_filter( $where, $query ) {
 			 * @param string $last_post_date @deprecated Last Post Date timestamp.
 			 */
 			$operator       = 'ASC' === $_REQUEST['query_args']['order'] ? '>' : '<'; // phpcs:ignore WordPress.Security.NonceVerification.Recommended, WordPress.Security.ValidatedSanitizedInput.InputNotValidated -- no changes to the site.
```

Comment:
I'm not sure this assign makes sense anymore since in line 756 we return early if `'DESC' !== $_REQUEST['query_args']['order']` 
@jeherve I see you have been working with Infinite Scroll in the past. Is line 756 a bug or should we directly do `$operator = '<'` ?

Reply:
It looks like the $operator and $last_post_date were supposed to be deprecated.
https://github.com/Automattic/jetpack/blob/d9973a16de729654cf694bf62df7ec5f16150eb5/projects/plugins/jetpack/modules/infinite-scroll/infinity.php#L779-L780

Reply:
Yes, I don't know that this parameter is needed at all anymore. It seems like it would be safe to remove, maybe using `apply_filters_deprecated()`

Reply:
> Yes, I don't know that this parameter is needed at all anymore. It seems like it would be safe to remove, maybe using `apply_filters_deprecated()`

TBH I don't even like the filter since it is supposed to be something similar to [posts_where](https://developer.wordpress.org/reference/hooks/posts_where/) but it works differently since it adds to the `$where` and doesn't replace it the way `posts_where` does.

By checking the [official documentation](https://developer.wordpress.org/reference/functions/apply_filters_deprecated/), it looks like we would need to replace it with a new filter (`$replacement`), but the `$args` are the same ones for both new and replacement filters which defeats the purpose of removing `$operator`.

Would something like the below diff work? I have not added a replacement purposefully since the idea is to remove the filter and the fields entirely in a future release.

```diff
diff --git a/projects/plugins/jetpack/modules/infinite-scroll/infinity.php b/projects/plugins/jetpack/modules/infinite-scroll/infinity.php
index 52929d303b..f6a2c3f5b0 100644
--- a/projects/plugins/jetpack/modules/infinite-scroll/infinity.php
+++ b/projects/plugins/jetpack/modules/infinite-scroll/infinity.php
@@ -778,9 +778,9 @@ class The_Neverending_Home_Page {
                         * @param string $operator @deprecated Query operator.
                         * @param string $last_post_date @deprecated Last Post Date timestamp.
                         */
-                       $operator       = 'ASC' === $_REQUEST['query_args']['order'] ? '>' : '<'; // phpcs:ignore WordPress.Security.NonceVerification.Recommended, WordPress.Security.ValidatedSanitizedInput.InputNotValidated -- no changes to the site.
-                       $last_post_date = sanitize_text_field( wp_unslash( $_REQUEST['last_post_date'] ) ); // phpcs:ignore WordPress.Security.NonceVerification.Recommended, WordPress.Security.ValidatedSanitizedInput.InputNotValidated -- no changes to the site.
-                       $where         .= apply_filters( 'infinite_scroll_posts_where', $clause, $query, $operator, $last_post_date );
+                       $operator       = '<';
+                       $last_post_date = isset( $_REQUEST['last_post_date'] ) ? sanitize_text_field( wp_unslash( $_REQUEST['last_post_date'] ) ) : ''; // phpcs:ignore WordPress.Security.NonceVerification.Recommended -- no changes to the site
+                       $where         .= apply_filters_deprecated( 'infinite_scroll_posts_where', array( $clause, $query, $operator, $last_post_date ), '$$next-version$$', '' );
                }
```

Reply:
Yeah, I think that would be okay. 🚢

Reply:
Implemented in [3b49148](https://github.com/Automattic/jetpack/pull/39618/commits/3b49148834a54e8982d5e0dce82ee436f930e5a9)

Updated the test instructions too